nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-processor-details.js

```diff
@@ -179,6 +179,9 @@
                 readOnly: true,
                 getParameterContext: function (groupId) {
                     // processors being configured must be in the current group
+                    if(typeof nfCanvasUtils === "undefined"){
```

Comment:
Rather than conditionally returning null, can we update this to conditional populate `getParameterContext`. This would offer a minor performance improvement as the `getParameterContext` configuration option is optional. By only populating `getParameterContext` when `nfCanvasUtils` is available we will avoid two calls into `getParameterContext` for every row in the table.

Reply:
Thanks for the suggestions, handled to set the getParameterContext when the nfCanvasUtils is available. regarding the passing of nfCanvasUtil and injection of it , i agree we can make it consistent across, will try to work on it with a new PR